staticviews should return 404 errors when templates aren't found

visit an invalid url like /uber/static_views/stafferCo

expected: error 404: stafferCo not found

actual:  500 server error:
```
Location: super2017.uber.magfest.org    Time: 2016-08-18T08:27:23.062Z
https://docker.uber.magfest.org/app/kibana#/doc/filebeat-*/filebeat-2016.08.18/log?id=AVacww8zz9OUXxfDi6ML
2016-08-18 04:27:22,612 [ERROR] uber.server: Exception encountered
   Request: GET /uber/static_views/stafferCo HTTP/1.0
   Current admin user is: [non-admin user]
   Request Params:
   
   Session Params:
   dict_items([])
   Request Headers:
     ACCEPT-ENCODING: gzip,deflate,br
     CONNECTION: close
     ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
     Remote-Addr: 127.0.0.1
     X-REAL-IP: 66.249.65.181
     FROM: googlebot(at)googlebot.com
     HOST: super2017.uber.magfest.org
     USER-AGENT: Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)
     X-FORWARDED-FOR: 66.249.65.181
   Traceback (most recent call last):
     File "/usr/local/uber/plugins/uber/uber/decorators.py", line 232, in render
       template = loader.select_template(listify(template_name_list))
     File "/usr/local/uber/env/lib/python3.4/site-packages/django/template/loader.py", line 186, in select_template
       raise TemplateDoesNotExist(', '.join(not_found))
   django.template.base.TemplateDoesNotExist: static_views/stafferCo
   
   During handling of the above exception, another exception occurred:
   
   Traceback (most recent call last):
     File "/usr/local/uber/env/lib/python3.4/site-packages/cherrypy/_cprequest.py", line 670, in respond
       response.body = self.handler()
     File "/usr/local/uber/env/lib/python3.4/site-packages/cherrypy/lib/encoding.py", line 217, in __call__
       self.body = self.oldhandler(*args, **kwargs)
     File "/usr/local/uber/env/lib/python3.4/site-packages/cherrypy/_cpdispatch.py", line 61, in __call__
       return self.callable(*self.args, **self.kwargs)
     File "/usr/local/uber/plugins/uber/uber/server.py", line 55, in default
       content = render(template_name)
     File "/usr/local/uber/plugins/uber/uber/decorators.py", line 236, in render
       django_template_source_info = getattr(e, 'django_template_source')
   AttributeError: 'TemplateDoesNotExist' object has no attribute 'django_template_source' 
```

I keep seeing this trip our logs as a 500 server error when googlebot tries to index (for some reason) a non-existent template

--- a/uber/decorators.py
+++ b/uber/decorators.py
@@ -231,6 +231,8 @@ def render(template_name_list, data=None):
     try:
         template = loader.select_template(listify(template_name_list))
         rendered = template.render(Context(data))
+    except django.template.base.TemplateDoesNotExist:
+        raise
     except Exception as e:
         source_template_name = '[unknown]'
         django_template_source_info = getattr(e, 'django_template_source')

--- a/uber/server.py
+++ b/uber/server.py
@@ -52,7 +52,10 @@ class StaticViews:
         content_filename = self.get_filename_from_path_args(path_args)
 
         template_name = self.get_full_path_from_path_args(path_args)
-        content = render(template_name)
+        try:
+            content = render(template_name)
+        except django.template.base.TemplateDoesNotExist as e:
+            raise cherrypy.HTTPError(404, "Couldn't find {}".format(template_name)) from e
 
         guessed_content_type = mimetypes.guess_type(content_filename)[0]
         return cherrypy.lib.static.serve_fileobj(content, name=content_filename, content_type=guessed_content_type)

